cmd/dp-deployer/main.go

```diff
@@ -117,7 +116,33 @@ func main() {
 	case <-ctx.Done():
 		log.Event(ctx, "context done", log.INFO)
 	}
-	wg.Wait()
+
+	log.Event(ctx, fmt.Sprintf("shutdown with timeout: %s", cfg.GracefulShutdownTimeout), log.INFO)
+	shutdownContext, cancel := context.WithTimeout(context.Background(), cfg.GracefulShutdownTimeout)
+
+	go func() {
+
+		// Shutdown HTTP server
+		log.Event(shutdownContext, "closing http server", log.INFO)
+		if err := httpServer.Shutdown(ctx); err != nil {
+			log.Event(shutdownContext, "failed to gracefully close http server", log.ERROR, log.Error(err))
+		}
+		log.Event(ctx, "http server gracefully closed ", log.INFO)
+
+		// Stop healthcheck
+		hc.Stop()
+
+		e.Close()
+
+		<-shutdownContext.Done()
+		if shutdownContext.Err() == context.DeadlineExceeded {
+			log.Event(shutdownContext, "shutdown timeout", log.ERROR, log.Error(shutdownContext.Err()))
+		} else {
+			log.Event(shutdownContext, "done shutdown gracefully", log.ERROR, log.Data{"context": shutdownContext.Err()})
+		}
+		os.Exit(1)
```

Comment:
I think this should be inside the `if` block? I think we only want to exit with a code 1 if something went wrong.